### PR TITLE
Improve overall testing posture and fix docker login issue

### PR DESCRIPTION
**Pull request by sonnysideup**
This PR is somewhat of jumbled mess (sorry).

## Changes

- Adds real-deal-banana-peel e2e testing
- Removes all of the unnecessary kustomize garbage
- Cleans up Makefile
- Fixes the issue with image expiry on tagged release images
- Adds unit tests to message publishing bits
- Adds docker login capability so forge can pull images `FROM` private registries (this is coupled to the registry push credentials for now)

### CODE REVIEW

File: internal/credentials/login.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	if dockerCfg.CredentialHelpers[serverAddr] != "" && authConfig.Username != "" && authConfig.Password != "" {
+		return dockerCfg, &authConfig, nil
+	}
+	authConfig.ServerAddress = serverAddr
+	authConfig.Username = username
+	authConfig.Password = password
+
+	return dockerCfg, &authConfig, nil
+}
+
```
Comment (steved) [inline internal/credentials/login.go:75]: I can't quite tell, but is this more "native" with https://github.com/moby/buildkit/pull/1397? I don't love the idea of writing out a docker config.

File: e2e/run-test.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  if [[ $state == "Completed" ]]; then
+    info "Test succeeded"
+    break
+  fi
+
+    if [[ $state == "Failed" ]]; then
+    error "Test failed"
+    kubectl logs --namespace "$namespace" --selector app.kubernetes.io/name=forge
+    break
```
Comment (steved) [inline e2e/run-test.sh:132]: I might be dense, but how does this cause a test "failure"?

Comment (sonnysideup) [reply e2e/run-test.sh:132]: Nope. You're right, good catch.

File: internal/credentials/login.go (hunk 1 above)
Comment (sonnysideup) [reply internal/credentials/login.go:75]: Yeah, but we don't have this available with our current versions of moby/containerd unfortunately. We can convert when we upgrade those deps. Make sense?

**Pull request merged**